### PR TITLE
cache/redis: use redisConfig to receive incoming JSON (previously using a map)

### PR DESCRIPTION
**Pull request by Handkerchiefs-t**
- use redisConfig to receive incoming JSON (previously using a map)
- change "timeout" to "idleTimeout"

### CODE REVIEW

File: client/cache/redis/redis.go [hunk 1]
```diff
@@ -43,8 +42,14 @@ import (
...
 
-// DefaultKey defines the collection name of redis for the cache adapter.
-var DefaultKey = "beecacheRedis"
+const (
+	// DefaultKey defines the collection name of redis for the cache adapter.
+	DefaultKey = "beecacheRedis"
+	// DefaultMaxIdle defines the default max idle connection number.
+	DefaultMaxIdle = 3
+	// DefaultIdleTimeout defines the default idleTimeout .
+	DefaultIdleTimeout = time.Second * 180
```
Comment (flycash) [inline client/cache/redis/redis.go:51]: make these two as private:
```go
defaultMaxIdle = 3
defaultIdleTimeout = time.Second * 180
```

File: client/cache/redis/redis.go [hunk 2]
```diff
@@ -204,60 +209,93 @@ func (rc *Cache) Scan(pattern string) (keys []string, err error) {
...
+	Key             string `json:"key"`
 	// Format redis://<password>@<host>:<port>
-	cf["conn"] = strings.Replace(cf["conn"], "redis://", "", 1)
-	if i := strings.Index(cf["conn"], "@"); i > -1 {
-		cf["password"] = cf["conn"][0:i]
-		cf["conn"] = cf["conn"][i+1:]
-	}
+	Conn           string `json:"conn"`
+	MaxIdle        int    `json:"maxIdle"`
+	IdleTimeoutStr string `json:"timeout"`
```
Comment (flycash) [inline client/cache/redis/redis.go:262]: Timeout time.Duration `json:"timeout"`

File: client/cache/redis/redis.go [hunk 3]
```diff
@@ -204,60 +209,93 @@ func (rc *Cache) Scan(pattern string) (keys []string, err error) {
...
+
+	// parse from Conn
+	password string
+	// parse from IdleTimeoutStr
+	idleTimeout time.Duration
```
Comment (flycash) [inline client/cache/redis/redis.go:267]: while use `Timeout time.Duration` you don't need this field, and please remove it

File: client/cache/redis/redis.go (hunk 2 above)
Comment (flycash) [reply client/cache/redis/redis.go:262]: Add comment: `Timeout used for idle connection`

**Review by flycash — CHANGES_REQUESTED** [review]
Put this merge request to CHANGELOG file

**Comment by Handkerchiefs-t** [comment]
@flycash 
I made the following modifications to the code:
1. Use the string type to receive the config and convert them in the **redisConfig.parse** method.
2. Still use the string type to receive the **timeout**.
3. Set **defaultMaxIdle** and **defaultTimeout** as private.

**Pull request merged**